### PR TITLE
dash/docs-highcharts-component

### PR DESCRIPTION
**Pull request by sebastianbochan**
Created additonal docs of the Highcharts Component.

### CODE REVIEW

**Review by Vazonik — APPROVED** [review]
Looks good!

File: docs/dashboards/highcharts-component.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+<script src="https://code.highcharts.com/dashboards/modules/dashboards-plugin.js"></script>
+```
+
+2. The Highcharts Component uses [styledMode](https://api.highcharts.com/highcharts/chart.styledMode) by default, so you need to load also the set of CSS styles to display Highcharts properly.
+```css
+@import url("https://code.highcharts.com/css/highcharts.css");
+```
+More information about styling charts, you can find in our [docs](https://www.highcharts.com/docs/chart-design-and-style/style-by-css).
+
+3. Now you have loaded all required files, so you should define the cell `cell: 'dashboard-col-0'`, where the component should be created.
```
Comment (karolkolodziej) [inline docs/dashboards/highcharts-component.md:25]: It doesn't have to be `dashboard-col-0'. I would rephrase it like: 

`After loading the necessary files, define a cell using a unique identifier for  example cell: 'dashboard-col-0'`

File: docs/dashboards/highcharts-component.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                data: [1, 2, 3, 4]
+            }]
+        }
+    }]
+});
+```
+
+## Working with data
+You can either define static data, as you would do in the basic highcharts chart, or use the [dataPool](https://www.highcharts.com/docs/dashboards/data-handling) to connect some dynamic data. The data gets parsed through the `columnAssignment` option to map correct values from the connector to reflect them in the series.
+[Here is the example](https://www.highcharts.com/samples/embed/dashboards/components/component-highcharts). The null value keeps columns selectively out of the chart. If data connector is connected, you can load the Highcharts' `dragDrop` module, to allow the user to change the value and sync the changes of this value with other components. Also, the editing is disabled by default, if the series data is based on the columns in the connector, which were created by `mathModifier`. You can read more in the `dataPool` section.
```
Comment (karolkolodziej) [inline docs/dashboards/highcharts-component.md:66]: Please add a code snippet here on how to connect data connector with HCC.

**Review by karolkolodziej — CHANGES_REQUESTED** [review]

**Comment by karolkolodziej** [comment]
I'm also wandering weather we should replace the component description we have in `types-of-components.md` with links to this documentation.

File: docs/dashboards/highcharts-component.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+```html
+<script src="https://code.highcharts.com/dashboards/dashboards.js"></script>
+<script src="https://code.highcharts.com/highcharts.js"></script>
+<script src="https://code.highcharts.com/dashboards/modules/dashboards-plugin.js"></script>
+```
+
+2. The Highcharts Component uses [styledMode](https://api.highcharts.com/highcharts/chart.styledMode) by default, so you need to load also the set of CSS styles to display Highcharts properly.
+```css
+@import url("https://code.highcharts.com/css/highcharts.css");
```
[the remark this answers is not in the document]
Comment (sebastianbochan) [reply docs/dashboards/highcharts-component.md:21]: I agree

File: docs/dashboards/highcharts-component.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+===
+
+The Highcharts Component allows the end-user to define a chart in the dashboard. Charts are most often used to visualize data that changes over time.
+
+<iframe style="width: 100%; height: 470px; border: none;" src=https://www.highcharts.com/samples/embed/dashboards/components/component-highcharts allow="fullscreen"></iframe>
+
+### How to start
+To get started quickly we need to load the JavaScript and CSS files in the following order.
+
+1. To be able to use Highcharts Component you first have to load Highcharts [Highcharts](https://code.highcharts.com/highcharts.js) as usual and load the additional [Dashboards plugin](https://code.highcharts.com/dashboards/modules/dashboards-plugin.js).
```
Comment (goransle) [inline docs/dashboards/highcharts-component.md:11]: Double Highcharts might not be intended?

File: docs/dashboards/highcharts-component.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+```html
+<script src="https://code.highcharts.com/dashboards/dashboards.js"></script>
+<script src="https://code.highcharts.com/highcharts.js"></script>
+<script src="https://code.highcharts.com/dashboards/modules/dashboards-plugin.js"></script>
+```
+
+Alternatively, you can also use package, loaded via NPM.
```
Comment (goransle) [inline docs/dashboards/highcharts-component.md:19]: Suggestion: "... the package loaded from NPM", or just "the NPM package"

File: docs/dashboards/highcharts-component.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+```
+
+2. The Highcharts Component uses [styledMode](https://api.highcharts.com/highcharts/chart.styledMode) by default, so you need to load also the set of CSS styles to display Highcharts properly.
+```css
+@import url("https://code.highcharts.com/css/dashboards.css");
+@import url("https://code.highcharts.com/css/highcharts.css");
+```
+More information about styling charts, you can find in our [docs](https://www.highcharts.com/docs/chart-design-and-style/style-by-css).
+
+3. After loading the necessary files, define a cell using a unique identifier for example cell: `dashboard-col-0`.
```
Comment (goransle) [inline docs/dashboards/highcharts-component.md:42]: Maybe a link to https://www.highcharts.com/docs/dashboards/your-first-dashboard here, in case the reader is not familiar with how to define a cell?

File: docs/dashboards/highcharts-component.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
+Highcharts Component
+===
+
+The Highcharts Component allows the end-user to define a chart in the dashboard. Charts are most often used to visualize data that changes over time.
+
+<iframe style="width: 100%; height: 470px; border: none;" src=https://www.highcharts.com/samples/embed/dashboards/components/component-highcharts allow="fullscreen"></iframe>
+
+### How to start
```
Comment (goransle) [inline docs/dashboards/highcharts-component.md:8]: This should be a `##`

**Review by goransle — APPROVED** [review]

**Review by karolkolodziej — APPROVED** [review]

File: docs/dashboards/highcharts-component.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+```
+
+2. The Highcharts Component uses [styledMode](https://api.highcharts.com/highcharts/chart.styledMode) by default, so you need to load also the set of CSS styles to display Highcharts properly.
+```css
+@import url("https://code.highcharts.com/css/dashboards.css");
+@import url("https://code.highcharts.com/css/highcharts.css");
+```
+More information about styling charts, you can find in our [docs](https://www.highcharts.com/docs/chart-design-and-style/style-by-css).
+
+3. After loading the necessary files, define a cell using a unique identifier for example cell: `dashboard-col-0`.
```
Comment (pawellysy) [inline docs/dashboards/highcharts-component.md:42]:
```suggestion
3. After loading the necessary files, define a cell using a unique identifier for example `cell: dashboard-col-0`.
```

File: docs/dashboards/highcharts-component.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+
+You can find more information how to create a layout in dashboard [here](https://www.highcharts.com/docs/dashboards/your-first-dashboard).
+
+4. Declare the [type of chart](https://www.highcharts.com/docs/chart-and-series-types/chart-types) and all of the charts options in the `chartOptions` object.
```
Comment (pawellysy) [inline docs/dashboards/highcharts-component.md:46]: You don't really need to declare the type of the chart explicitly, because if you won't do it, nothing will happen, so maybe just leave out the `Declare all of the chart options in the chartOptions object` ?

**Review by pawellysy — CHANGES_REQUESTED** [review]

File: docs/dashboards/highcharts-component.md (hunk 8 above)
Comment (sebastianbochan) [reply docs/dashboards/highcharts-component.md:42]: Missing quotes, so I will commit it on my own.

File: docs/dashboards/highcharts-component.md (hunk 9 above)
Comment (sebastianbochan) [reply docs/dashboards/highcharts-component.md:46]: The line chart is the default, so the chart's type is not necessary there. Makes sense.

**Review by pawellysy — APPROVED** [review]
